packages/generator/tests/acceptance/test_pin_aware_measurement_client.py

```diff
@@ -0,0 +1,197 @@
+import importlib.util
+import pathlib
+from types import ModuleType
+from typing import Generator
+
+import grpc
+import pytest
+from ni_measurement_plugin_sdk_service.measurement.service import MeasurementService
+from ni_measurement_plugin_sdk_service.session_management import PinMapContext
+
+from ni_measurement_plugin_sdk_generator.client import create_client
+from tests.utilities.discovery_service_process import DiscoveryServiceProcess
+from tests.utilities.measurements import pin_aware_measurement
+
+
+def test___measurement_plugin_client___measure_with_pin_map_registration___returns_output(
+    measurement_plugin_client_module: ModuleType,
+    pin_map_directory: pathlib.Path,
+) -> None:
+    pin_map_path = pin_map_directory / "1Smu1ChannelGroup1Pin1Site.pinmap"
+    output_type = getattr(measurement_plugin_client_module, "Output")
+    expected_output = output_type(
+        pin_map_id=str(pin_map_path),
+        sites=[0],
+        session_names=["DCPower1/0"],
+        resource_names=["DCPower1/0"],
+        channel_lists=["DCPower1/0"],
+    )
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+
+    measurement_plugin_client.register_pin_map(pin_map_path)
+    output = measurement_plugin_client.measure()
+
+    assert output == expected_output
+
+
+def test___measurement_plugin_client___measure_without_pin_map_registration___raises_no_sessions_error(
+    measurement_plugin_client_module: ModuleType,
+) -> None:
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+
+    with pytest.raises(grpc.RpcError) as exc_info:
+        _ = measurement_plugin_client.measure()
+
+    assert exc_info.value.code() == grpc.StatusCode.UNKNOWN
+    assert "No sessions reserved." in (exc_info.value.details() or "")
+
+
+def test___measurement_plugin_client___register_pin_map_without_pin_map_context___updates_pin_map_id_and_sites(
```

Comment:
Should the "then" phrase say `...___creates_pin_map_context_with_pin_map_id_and_sites` because we're asserting the pin map context object?

Or 

Should you only assert the pin map id and sites values separately?

Reply:
Checking the `pin_map_context` is the m=aim of this test so updated the test case name.